bindings/matrix-sdk-ffi/src/room/mod.rs

```diff
@@ -1139,6 +1139,59 @@ impl Room {
 
         Ok(Arc::new(RoomPreview::new(AsyncRuntimeDropped::new(client), room_preview)))
     }
+
+    /// Toggle a MSC4306 subscription to a thread in this room, based on the
+    /// thread root event id.
+    ///
+    /// If `subscribed` is `true`, it will subscribe to the thread, with a
+    /// precision that the subscription was manually requested by the user
+    /// (i.e. not automatic).
+    ///
+    /// If the thread was already subscribed to (resp. unsubscribed from), while
+    /// trying to subscribe to it (resp. unsubscribe from it), it will do
+    /// nothing, i.e. subscribing (resp. unsubscribing) to a thread is an
+    /// idempotent operation.
+    pub async fn toggle_thread_subscription(
+        &self,
+        thread_root_event_id: String,
+        subscribed: bool,
```

Comment:
Why are we not providing a way to say whether this is automatic or not? Won't clients need to do that e.g. if the user is mentioned?

Reply:
By definition, the automatic subscriptions are caused by the SDK, not external clients; at least that's my current understanding of the MSC.

Reply:
Ah right, makes sense.